xmodule/library_sourced_block.py

```diff
@@ -38,50 +60,91 @@ class LibrarySourcedBlock(StudioEditableXBlockMixin, EditableChildrenMixin, XBlo
-        default="Library Sourced Content",
+        default="Library Reference Block",
         display_name=_("Display Name"),
         scope=Scope.content,
     )
+    source_library_id = String(
+        display_name=_("Library"),
+        help=_("Select the library from which you want to draw content."),
+        scope=Scope.settings,
+        values_provider=lambda instance: instance.source_library_values(),
+    )
+    source_library_version = String(
+        # This is a hidden field that stores the version of source_library when we last pulled content from it
+        display_name=_("Library Version"),
+        scope=Scope.settings,
+    )
     source_block_ids = List(
         display_name=_("Library Blocks List"),
         help=_("Enter the IDs of the library XBlocks that you wish to use."),
         scope=Scope.content,
     )
-    editable_fields = ("display_name", "source_block_ids")
+    editable_fields = ("source_block_ids",)
     has_children = True
     has_author_view = True
+
     resources_dir = 'assets/library_source_block'
-    MAX_BLOCKS_ALLOWED = 10
+
+    preview_view_js = {
+        'js': [],
+        'xmodule_js': resource_string(__name__, 'js/src/xmodule.js'),
+    }
+    preview_view_css = {
+        'scss': [],
+    }
+
+    mako_template = 'widgets/metadata-edit.html'
+    studio_js_module_name = "VerticalDescriptor"
+    studio_view_js = {
+        'js': [
+            resource_string(__name__, 'js/src/vertical/edit.js'),
+        ],
+        'xmodule_js': resource_string(__name__, 'js/src/xmodule.js'),
+    }
+    studio_view_css = {
+        'scss': [],
+    }
 
     def __str__(self):
         return f"LibrarySourcedBlock: {self.display_name}"
 
-    def __init__(self, *args, **kwargs):
-        super().__init__(*args, **kwargs)
-        if not self.source_block_ids:
-            self.has_children = False
+    def render_children(self, context, fragment, can_reorder=False, can_add=False):
+        """
+        Renders the children of the module with HTML appropriate for Studio. If can_reorder is True,
+        then the children will be rendered to support drag and drop.
+        """
+        contents = []
+        for child in self.get_children():  # pylint: disable=no-member
+            if can_reorder:
+                context['reorderable_items'].add(child.location)
+            context['can_add'] = can_add
+            context['is_selected'] = str(child.location) in self.source_block_ids
+            rendered_child = child.render(EditableChildrenMixin.get_preview_view_name(child), context)
+            fragment.add_fragment_resources(rendered_child)
+
+            contents.append({
+                'id': str(child.location),
+                'content': rendered_child.content
+            })
 
-    def studio_view(self, context):
+        fragment.add_content(self.runtime.service(self, 'mako').render_template("studio_render_children_view.html", {  # pylint: disable=no-member
```

Comment:
```suggestion
        fragment.add_content(self.runtime.service(self, 'mako').render_lms_template("studio_render_children_view.html", {  # pylint: disable=no-member
```
Recent change: Use `render_lms_template` or `render_cms_template` instead of `render_template` if you have to use a mako template. Despite the name, this template is found in `lms/templates/` so is an LMS template.

Reply:
https://github.com/openedx/edx-platform/pull/33263#pullrequestreview-1627813331